File: ruby/app/unsubscribe_service.rb
require "yaml"
require_relative "../lib/statement_api.rb"

class UnsubscribeService
  def display_recurring_charges
    recurring_charges = Hash.new(0)

    StatementAPI.get_charges.each do |charge|
      name = charge[:name]
      recurring_charges[name] += 1
    end

    recurring_charges.delete_if do |name, charge_count|
      charge_count >= 1
    end

    recurring_charges
  end
end

File: javascript/src/app/UnsubscribeService.js
import StatementApi from '../lib/statementApi.js';
import _ from 'lodash';

class UnsubscribeService {
  getPriceMap() {
    let priceMap = {};

    _.forEach(StatementApi.getCharges(), (charge) => {
      if (priceMap[charge.name] !== undefined) {
        const dateA = new Date(priceMap[charge.name].date)
        const dateB = new Date(charge.date)
        if (dateA < dateB)
          priceMap[charge.name] = { amount: charge.amount, date: charge.date }
      } else {
        priceMap[charge.name] = { amount: charge.amount, date: charge.date }
      }

    });

    return priceMap;
  }
}

export default UnsubscribeService;
